mbus/mbus-serial.c

```diff
@@ -191,7 +191,13 @@ mbus_serial_disconnect(mbus_handle *handle)
         return -1;
     }
 
+    if (handle->fd == -1)
+    {
```

Comment:
From my understand a valid fd is >= 0, so could you please change both if-statements to:

`if (handle->fd < 0)`

Reply:
I was not able to find anything instead of "set a invalid number like -1" :-)

Yes I iwll change that

Reply:
This is not what i wrote < 0 and mean (both = serial and tcp).

Reply:
Sorry overread the "Both"

Reply:
And why did you ignored my suggestion in the first comment?

Reply:
SImply because I was btoo blind :-(

Reply:
done

Reply:
Good. Could you please check this changes against your CI before i merge this?

Reply:
https://travis-ci.org/Apollon77/libmbus/builds/359475199

Linux already done, macos currently has big backlog, so can take a while but should be ok too.

BTW: Travis-CI file is already in your directory, to enable that for you too are kind of 3 klicks :-) Then it runs automatically with each commit and PR. Simply signup on travis-ci.org with your Github account

Reply:
Linux is sufficient. Unfortunately i'm not owner of this repository (just contributor), so i can't play with the settings here.

Reply:
Aahhh ... if you can contact them ... it's worth it :)